### PR TITLE
Conversion of upstream Rook docs (JIRA #SES-1966)

### PR DESCRIPTION
**Pull request by lproven**
Work in progress at the moment: Almost all files have been edited to include basic entities, and formatted the XML to be more inline with SUSE practices. 

This pull request does not go any further than the scope of simply adding the files, and making the basic edits. All files will require full editorial assistance once this pull request has been merged successfully. 

Signed-off-by: Alexandra Settle <asettle@suse.com>

### CODE REVIEW

**Comment by asettle** [comment]
@BlaineEXE  and @ml8mr  - [book-storage-rook_color_en.pdf](https://github.com/SUSE/doc-ses/files/5226078/book-storage-rook_color_en.pdf)

**Comment by asettle** [comment]
[book-storage-rook_color_en.pdf](https://github.com/SUSE/doc-ses/files/5226304/book-storage-rook_color_en.pdf)

File: xml/admin_caasp_advanced_config.xml [hunk 1]
```diff
@@ -5,13 +5,836 @@
...
+        <para>
+          <emphasis role="strong">This ClusterRole is needed no matter if
+            you want to use a RoleBinding per namespace or a
+            ClusterRoleBinding.</emphasis>
+        </para>
+        <screen>
+          apiVersion: rbac.authorization.k8s.io/v1
+          kind: ClusterRole
+          metadata:
+          name: rook-ceph-agent-mount
```
Comment (asettle) [inline xml/admin_caasp_advanced_config.xml:181]: @BlaineEXE what do we do with these examples?

**Comment by asettle** [comment]
[book-storage-rook_color_en.pdf](https://github.com/SUSE/doc-ses/files/5226623/book-storage-rook_color_en.pdf)

File: xml/admin_caasp_block.xml [hunk 2]
```diff
@@ -5,13 +5,261 @@
...
+    RBD</link> directory.
+   </para>
+  </note>
+
+  <para>
+   Save this <literal>StorageClass</literal> definition as
+   <literal>storageclass.yaml</literal>:
+  </para>
+
+<screen>
```
Comment (thehejik) [inline xml/admin_caasp_block.xml:62]: YAML examples are missing proper indentation. Please follow examples https://github.com/rook/rook/tree/master/cluster/examples/kubernetes/ceph

Also original document has missing ident. in almost every YAML. https://github.com/SUSE/doc-ses/files/5226623/book-storage-rook_color_en.pdf

Without the correct ident the examples cannot by used.

File: xml/admin_caasp_cephfs.xml [hunk 3]
```diff
@@ -5,13 +5,335 @@
...
+        the metadata pool with replication of three and a single data pool
+        with replication of three. For more options, see the documentation
+        on <link xlink:href="ceph-filesystem-crd.md">creating shared
+          filesystems</link>.
+      </para>
+      <para>
+        Save this shared filesystem definition as
+        <literal>filesystem.yaml</literal>:
+      </para>
+      <screen>
```
Comment (thehejik) [inline xml/admin_caasp_cephfs.xml:66]: dtto ident

File: xml/admin_caasp_cephfs.xml [hunk 4]
```diff
@@ -5,13 +5,335 @@
...
+          1.13 and newer. Examples are found in the
+          <link xlink:href="https://github.com/rook/rook/tree/%7B%7B%20branchName%20%7D%7D/cluster/examples/kubernetes/ceph/csi/cephfs">CSI
+            CephFS</link> directory.
+        </para>
+      </note>
+      <para>
+        Save this storage class definition as
+        <literal>storageclass.yaml</literal>:
+      </para>
+      <screen>
```
Comment (thehejik) [inline xml/admin_caasp_cephfs.xml:135]: dtto ident

File: xml/admin_caasp_cephfs.xml [hunk 5]
```diff
@@ -5,13 +5,335 @@
...
+      </note>
+    </section>
+    <section xml:id="rook-consume-the-shared-filesystem-k8s-registry-sample">
+      <title>Consume the Shared Filesystem: K8s Registry Sample</title>
+      <para>
+        As an example, we will start the kube-registry pod with the shared
+        filesystem as the backing store. Save the following spec as
+        <literal>kube-registry.yaml</literal>:
+      </para>
+      <screen>
```
Comment (thehejik) [inline xml/admin_caasp_cephfs.xml:209]: dtto ident

File: xml/admin_caasp_cephtoolbox.xml [hunk 6]
```diff
@@ -5,12 +5,244 @@
...
+    <section xml:id="interactive-toolbox">
+      <title>Interactive Toolbox</title>
+      <para>
+        The rook toolbox can run as a deployment in a Kubernetes cluster
+        where you can connect and run arbitrary Ceph commands.
+      </para>
+      <para>
+        Save the tools spec as <literal>toolbox.yaml</literal>:
+      </para>
+      <screen>
```
Comment (thehejik) [inline xml/admin_caasp_cephtoolbox.xml:48]: dtto ident

File: xml/admin_caasp_cephtoolbox.xml [hunk 7]
```diff
@@ -5,12 +5,244 @@
...
+        collect the results later from the logs, you can run a script as a
+        Kubernetes Job. The toolbox job will run a script that is embedded
+        in the job spec. The script has the full flexibility of a bash
+        script.
+      </para>
+      <para>
+        In this example, the <literal>ceph status</literal> command is
+        executed when the job is created.
+      </para>
+      <screen>
```
Comment (thehejik) [inline xml/admin_caasp_cephtoolbox.xml:173]: dtto ident

File: xml/admin_caasp_object_dashboard.xml [hunk 8]
```diff
@@ -5,12 +5,344 @@
...
+      <section xml:id="node-port">
+        <title>Node Port</title>
+        <para>
+          The simplest way to expose the service in minikube or similar
+          environment is using the NodePort to open a port on the VM that
+          can be accessed by the host. To create a service with the
+          NodePort, save this yaml as
+          <literal>dashboard-external-https.yaml</literal>.
+        </para>
+        <screen>
```
Comment (thehejik) [inline xml/admin_caasp_object_dashboard.xml:162]: dtto ident

File: xml/admin_caasp_object_dashboard.xml [hunk 9]
```diff
@@ -5,12 +5,344 @@
...
+            </para>
+          </listitem>
+          <listitem>
+            <para>
+              Tells the reverse proxy that the dashboard itself does not
+              have a valid certificate (it is self-signed)
+            </para>
+          </listitem>
+        </orderedlist>
+        <screen>
```
Comment (thehejik) [inline xml/admin_caasp_object_dashboard.xml:284]: dtto ident

File: xml/admin_caasp_block.xml (hunk 2 above)
Comment (asettle) [reply xml/admin_caasp_block.xml:62]: Thanks :) we're still working through the edits here. I'll make sure these are adjusted.

**Comment by asettle** [comment]
@thehejik  - most of the indents should have been solved now! [book-storage-rook_color_en.pdf](https://github.com/SUSE/doc-ses/files/5277484/book-storage-rook_color_en.pdf)

**Comment by thehejik** [comment]
> @thehejik - most of the indents should have been solved now! [book-storage-rook_color_en.pdf](https://github.com/SUSE/doc-ses/files/5277484/book-storage-rook_color_en.pdf)

Much better but there are still unresolved indents, see page 82 and following.

**Comment by thehejik** [comment]
Maybe this is out of scope of this PR but shouldn't we mention how to assign label to a node for his specific role in SES cluster? Without it my SES cluster deployment failed.

There is a mention in chapter 1.4.1.3 Ceph Cluster Configuration but I think it is not complete and we should expand it a bit more.

According to Denis Kondratenko and Stefan Haas users have to assign one of the following label/values to a desired node.

Label and its value (probably not complete list, I was using `any` for my worker nodes): 
`node-role.rook-ceph/cluster=( mon | mgr | osd | mon-mgr-osd | any )`

Example for all nodes:
`kubectl label nodes --all node-role.rook-ceph/cluster=any`

Example for single node:
`kubectl label node <node-name> node-role.rook-ceph/cluster=any`

**Comment by asettle** [comment]
> Maybe this is out of scope of this PR but shouldn't we mention how to assign label to a node for his specific role in SES cluster? Without it my SES cluster deployment failed.
> 
> There is a mention in chapter 1.4.1.3 Ceph Cluster Configuration but I think it is not complete and we should expand it a bit more.
> 
> According to Denis Kondratenko and Stefan Haas users have to assign one of the following label/values to a desired node.
> 
> Label and its value (probably not complete list, I was using `any` for my worker nodes):
> `node-role.rook-ceph/cluster=( mon | mgr | osd | mon-mgr-osd | any )`
> 
> Example for all nodes:
> `kubectl label nodes --all node-role.rook-ceph/cluster=any`
> 
> Example for single node:
> `kubectl label node <node-name> node-role.rook-ceph/cluster=any`

Hi @thehejik - appreciate the feedback, thank you! Unfortunately, this is completely out of scope for this PR at this time :) what we are working on in this request is to simply convert the documentation that is hosted at `rook.io/docs` into Docbook XML so we can publish at `doc.suse.com`. This is a time-confusing process as we need to clean up the code (as you've noted with the indentation marking) and also provide edits to ensure this is up to SUSE standard. 

We are tracking any/all issues for Rook docs here: https://github.com/SUSE/doc-ses/issues you will see we have quite a lot assigned for Rook: https://github.com/SUSE/doc-ses/issues/assigned/asettle

You are more than welcome to review your comments against these bugs, and either add suggestions to the pre-existing ones, or we'd like to encourage you to open your own bug reports in an effort to track your comments. Once this pull request is closed, your comments will no longer be monitored :) 

I hope this has been helpful, thanks for reviewing and taking the time to provide feedback.

Alex

**Comment by ml8mr** [comment]
> You are more than welcome to review your comments against these bugs, and either add suggestions to the pre-existing ones, or we'd like to encourage you to open your own bug reports in an effort to track your comments. Once this pull request is closed, your comments will no longer be monitored :)

@thehejik - I just entered issue https://github.com/SUSE/doc-ses/issues/611 on this issue. (We've tripped on this before and just realized it was not documented as well.) Thanks for reporting it!

**Pull request merged**